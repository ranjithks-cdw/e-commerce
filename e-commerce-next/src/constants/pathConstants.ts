const PATH_CONSTANTS = {
    HOME: {
        PATH: '/',
        TITLE: 'Home',
    },
    PRODUCTS: {
        PATH: '/products',
        TITLE: 'Products',
    },
    ORDERS: {
        PATH: '/orders',
        TITLE: 'Orders',
    },
    CART: {
        PATH: '/cart',
        TITLE: 'Cart',
    },
    PROFILE: {
        PATH: '/profile',
        TITLE: 'Profile',
    },
    LOGIN: {
        PATH: '/login',
        TITLE: 'Login',
    }
};

export default PATH_CONSTANTS;