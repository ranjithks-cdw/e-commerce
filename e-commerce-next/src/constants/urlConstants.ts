export const URLS = {
    FEATURES: process.env.FEATURES,
}