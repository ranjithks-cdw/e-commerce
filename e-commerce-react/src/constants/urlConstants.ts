export const URLS = {
    FEATURES: process.env.REACT_APP_FEATURES,
};