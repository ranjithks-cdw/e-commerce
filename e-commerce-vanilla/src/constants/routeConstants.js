export const APP_PATHS = {
    HOME: {
        TITLE: 'Home',
        PATH: '/',
    },
    PRODUCTS: {
        TITLE: 'Products',
        PATH: '/products',
    },
    PRODUCT_DESCRIPTION: {
        TITLE: 'Product Description',
        PATH: '/products/:productId',
    },
    CART: {
        TITLE: 'Cart',
        PATH: '/cart',
    },
    ORDERS: {
        TITLE: 'Orders',
        PATH: '/orders',
    },
    PROFILE: {
        TITLE: 'Profile',
        PATH: '/profile',
    },
    LOGIN: {
        TITLE: 'Login',
        PATH: '/login',
    },
    LOGOUT: {
        TITLE: 'Logout',
        PATH: '/logout',
    },
};
