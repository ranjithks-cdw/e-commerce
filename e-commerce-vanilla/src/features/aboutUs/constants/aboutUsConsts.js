export const AboutUsConstants = {
    title: 'About Us',
};