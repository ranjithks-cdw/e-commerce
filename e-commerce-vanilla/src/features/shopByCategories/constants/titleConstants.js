export const SBCFeatureTitles = {
    PRODUCT_CATEGORIES: 'Product Categories',
    VIEW_ALL: 'View All',
};