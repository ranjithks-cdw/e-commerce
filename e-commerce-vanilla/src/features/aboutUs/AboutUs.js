import { createElement } from "../../utils/htmlUtils.js";
import { AboutUsConstants } from "./constants/aboutUsConsts.js";

const AboutUs = {
    addComponent: () =>  addComponent(),
    removeComponent: () =>  removeComponent(),
};

const addComponent = () => {
    const component = createDOM();

    return component;
};

const removeComponent = () => {
    
};

const createDOM = async () => {
    const aboutUs = createElement('div', {
        className: 'about-us',
    });

    const sectionTitle = createElement('p', {
        className: 'section-title',
        textContent: AboutUsConstants.title,
    });

    const data = await getAboutUsData();

    const sectionContent = createElement('p', {
        className:  'section-content',
        textContent: data
    });

    aboutUs.appendChild(sectionTitle);
    aboutUs.appendChild(sectionContent);

    return aboutUs;
};


const getAboutUsData = async () => {
    const data = `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.`;

    return data;
};
 
export default AboutUs;