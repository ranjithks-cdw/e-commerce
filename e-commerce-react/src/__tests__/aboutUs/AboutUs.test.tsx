import React from "react";
import { render, waitFor, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import AboutUs from "../../features/aboutUs/AboutUs";
import { AboutUsConstants } from "../../features/aboutUs/constants/aboutUsConsts";

describe('About Us', () => {
    it("renders component", async () => {
        
        render(<AboutUs />);
        const titleText = await screen.findByText(AboutUsConstants.title);
        expect(titleText).toBeInTheDocument();
    })

    it("matches snapshot", () => {
        const container = renderer.create(<AboutUs />).toJSON;
        expect(container).toMatchSnapshot();
    })

    jest.resetAllMocks();
})