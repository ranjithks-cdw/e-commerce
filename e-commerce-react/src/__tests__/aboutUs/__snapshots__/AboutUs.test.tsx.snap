// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us matches snapshot 1`] = `[Function]`;
